core/templates/pages/exploration-editor-page/statistics-tab/statistics-tab.directive.ts

```diff
@@ -47,162 +38,105 @@ angular.module('oppia').directive('statisticsTab', [
+                  '/pages/exploration-editor-page/statistics-tab/templates/' +
+                  'state-stats-modal.template.html'),
+                styleUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+                  '/pages/exploration-editor-page/statistics-tab/templates/' +
+                  'state-stats-modal.template.css'),
+                backdrop: true,
+                resolve: {
+                  interactionArgs: () => state.interaction.customizationArgs,
+                  stateName: () => stateName,
+                  stateStats: () => this.expStats.getStateStats(stateName),
+                  visualizationsInfo: () => stats.visualizations_info,
                 },
-                visualizationsInfo: function() {
-                  return stats.visualizations_info;
-                }
-              },
-            }).result.then(
-              () => {
-                stateStatsModalIsOpen = false;
-              },
-              () => {
-                AlertsService.clearWarnings();
-                stateStatsModalIsOpen = false;
-              }));
+              }).result);
           };
 
-          ctrl.$onInit = function() {
-            $scope.$on('refreshStatisticsTab', function() {
-              ctrl.refreshExplorationStatistics(_EXPLORATION_STATS_VERSION_ALL);
-            });
-            ctrl.COMPLETION_RATE_CHART_OPTIONS = {
-              chartAreaWidth: 300,
-              colors: ['green', 'firebrick'],
-              height: 100,
-              legendPosition: 'right',
-              width: 500
-            };
-            ctrl.COMPLETION_RATE_PIE_CHART_OPTIONS = {
-              title: '',
-              left: 230,
-              pieHole: 0.6,
-              pieSliceTextStyleColor: 'black',
-              pieSliceBorderColor: 'black',
-              chartAreaWidth: 500,
-              colors: ['#008808', '#d8d8d8'],
-              height: 300,
-              legendPosition: 'right',
-              width: 600
+          this.$onInit = () => {
+            this.stateStatsModalIsOpen = false;
+            $scope.onClickStateInStatsGraph = (stateName: string) => {
+              if (!this.stateStatsModalIsOpen) {
+                this.stateStatsModalIsOpen = true;
+                openStateStatsModal(stateName).then(
+                  () => {
+                    this.stateStatsModalIsOpen = false;
+                  },
+                  () => {
+                    AlertsService.clearWarnings();
+                    this.stateStatsModalIsOpen = false;
+                  });
+              }
             };
-            ctrl.currentVersion = _EXPLORATION_STATS_VERSION_ALL;
-
-            ctrl.hasTabLoaded = false;
-
-            ctrl.explorationHasBeenVisited = false;
+            $scope.explorationHasBeenVisited = false;
+            $scope.$on('refreshStatisticsTab', refreshExplorationStatistics);
           };
-        }
-      ]
+        },
+      ],
```

Comment:
Just to check, are these trailing commas intentional like it is mentioned [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Trailing_commas)?

Reply:
Yes, specifically: "[...] makes version-control diffs cleaner and editing code might be less troublesome."

Reply:
That's really cool! Thanks @brianrodri. I'll keep this in mind while reviewing future PRs. Seems like a good pattern to have in our codebase.
/cc @oppia/core-maintainers